tests/test_cart.py

```diff
@@ -1,15 +1,231 @@
 from __future__ import unicode_literals
 
 import json
+from uuid import uuid4
 
 import pytest
 from babeldjango.templatetags.babel import currencyfmt
 from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpRequest
 from mock import MagicMock, Mock
+from satchless.item import InsufficientStock
 
-from saleor.cart import decorators, forms, utils
+from saleor.cart import forms, utils
+from saleor.cart.decorators import (find_and_assign_cart,
+                                    get_anonymous_cart_from_token,
+                                    get_cart_from_request,
+                                    get_or_create_anonymous_cart_from_token,
+                                    get_or_create_cart_from_request,
+                                    get_or_create_user_cart, get_user_cart)
 from saleor.cart.models import Cart
-from satchless.item import InsufficientStock
+
+
+@pytest.fixture()
+def test_user(django_user_model):
+    return django_user_model.objects.get_or_create(
+        email='test@example.com', defaults={'is_active': True,
+                                            'is_staff': True,
+                                            'is_superuser': True})[0]
+
+
+@pytest.fixture()
+def opened_anonymous_cart(test_user):
+    return Cart.objects.get_or_create(user=None, status=Cart.OPEN)[0]
+
+
+@pytest.fixture()
+def cancelled_anonymous_cart(test_user):
+    return Cart.objects.get_or_create(user=None, status=Cart.CANCELED)[0]
+
+
+@pytest.fixture()
+def opened_user_cart(test_user):
+    return Cart.objects.get_or_create(user=test_user, status=Cart.OPEN)[0]
+
+
+@pytest.fixture()
+def cancelled_user_cart(test_user):
+    return Cart.objects.get_or_create(user=test_user, status=Cart.CANCELED)[0]
+
+
+def test_get_or_create_anonymous_cart_from_token(opened_anonymous_cart,
```

Comment:
I wanted to test those getters on populated db to test if the are not not returning good cart by mistake.